src/main/java/org/jboss/ejb/protocol/remote/ProtocolV1ObjectResolver.java

```diff
@@ -60,7 +61,7 @@ public Object writeReplace(final Object original) {
             return Affinity.NONE;
         } else if (original == Affinity.LOCAL) {
             // Swap a local affinity with a node affinity with the name of this node
-            return nodeAffinity;
+            return new NodeAffinity(connection.getRemoteEndpointName());
```

Comment:
This seems reversed to me: we have a local affinity on the local system, we want to send it to the peer, so the affinity we send to the peer should refer back to me (which is what "nodeAffinity" is equal to).  With this change, the affinity refers to the peer, so they will receive a node affinity pointing to themselves.